### PR TITLE
Increase the request_timeout from default

### PR DESCRIPTION
**Pull request by rooftopcellist**
- As AWX gets overloaded with jobs, the response time of HTTP requests to the api/v2/jobs endpoint
- The result is that AnsibleJob objects launcht he job, then keep checking for the status of the job that was launch. The GET request times out, and it launches a new job, which complicates the problem further.
- This behavior is seen when createing 100+ AnsibleJob objects at once
- We should keep this HTTP request_timeout abnormally high until the resource operator grows a better way to handle this (smart checks for timeout errors, etc.)

### CODE REVIEW

**Review by dsavineau — CHANGES_REQUESTED** [review]
We need to update the minimal awx.awx ansible collection version since the request_timeout parameter has been added in awx 22.7.0

Also would it make sense to make it a global parameter at CRD level so one hitting the hardcoded 600 seconds timeout could increase it in the CR ?

https://github.com/ansible/awx/commit/abc56305cc7ab0feb38c5b8c786f015ee7ed8b82

**Comment by rooftopcellist** [comment]
@dsavineau That is a good idea.  I just added the ability to configure request_timeout on the job and workflow CR's.  I also added a better description for what Job TTL is while there.  

I bumped the minimum version of awx.awx collection to 22.7.0, which is the version that option was introduced in.

File: config/crd/bases/tower.ansible.com_ansiblejobs.yaml [hunk 1]
```diff
@@ -40,6 +40,8 @@ spec:
                 type: string
               workflow_template_name:
                 type: string
+              request_timeout:
+                type: integer
```
Comment (dsavineau) [inline config/crd/bases/tower.ansible.com_ansiblejobs.yaml:44]: Shouldn't we add a description to this too?

File: config/crd/bases/tower.ansible.com_ansibleworkflows.yaml [hunk 2]
```diff
@@ -38,6 +38,8 @@ spec:
                 type: string
               workflow_template_name:
                 type: string
+              request_timeout:
+                type: integer
```
Comment (dsavineau) [inline config/crd/bases/tower.ansible.com_ansibleworkflows.yaml:42]: ditto

File: config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -61,6 +62,11 @@ spec:
         x-descriptors:
         - urn:alm:descriptor:com.tectonic.ui:text
         - urn:alm:descriptor:com.tectonic.ui:advanced
+      - displayName: Request Timeout
+        description: Time in seconds to wait for a response from the AWX API
+        path: request_timeout
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:text
```
Comment (dsavineau) [inline config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml:69]: maybe add `urn:alm:descriptor:com.tectonic.ui:advanced` too ?
What do you think ?

File: config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -198,10 +204,16 @@ spec:
...
+        description: Time to live for k8s job object after the playbook run has finished
         path: workflow_ttl
         x-descriptors:
         - urn:alm:descriptor:com.tectonic.ui:text
         - urn:alm:descriptor:com.tectonic.ui:advanced
+      - displayName: Request Timeout
+        description: Time in seconds to wait for a response from the AWX API
+        path: request_timeout
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:text
```
Comment (dsavineau) [inline config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml:217]: ditto

File: roles/job_runner/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+
+# Time before request to AWX api/v2/jobs/{id}/stdout/ times out (in seconds)
+request_timeout: 600
```
Comment (dsavineau) [inline roles/job_runner/defaults/main.yml:4]: I think we need this default value in other roles too right ? the current parameter is added for ansiblejob and ansibleworkflow but those are using job and workflow roles rather tha job_runner if I'm not mistaken

Comment (rooftopcellist) [reply roles/job_runner/defaults/main.yml:4]: @dsavineau The ansiblejob and ansibleworkflow roles actually both call the job_runner role to create a k8s pod to run the playbook in.

We only have this issue with the job_wait tasks that check if the job is complete or not.  There are only 3 references, and they are all in the job_runner role:
* https://github.com/search?q=repo%3Aansible%2Fawx-resource-operator%20job_wait&type=code

File: config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (rooftopcellist) [reply config/manifests/bases/awx-resource-operator.clusterserviceversion.yaml:69]: agreed

**Review by dsavineau — APPROVED** [review]

**Pull request merged**